src/main/java/com/bookshop/exception/BookNotFoundException.java

```diff
@@ -0,0 +1,7 @@
+package com.bookshop.exception;
+
+public class BookNotFoundException extends RuntimeException {

```

Comment:
Don’t customize your Exceptions in this case you need to create exceptions for each entity.
```suggestion
public class EntityNotFoundException extends RuntimeException {

```